### DISCUSSION TITLE
Error when using calcGrains

### DISCUSSION THREAD

**Discussion by rtaufner** [mtex-toolbox/mtex#600] [Ask Anything]
Hi Ralf and colleagues, 

I've been trying to use the latest version of Mtex 5.5.beta.6 (I was using Mtex 5.1 before), but I keep getting an error when calculating grains. This same problem was posted on the [old forum](https://groups.google.com/g/mtexmail/c/B_AKJnHXwwI/m/UBbSzFZ_CQAJ) a while ago. At that time, you recommended relaunching the MatLab or reinstall Mtex. I've tried both ways and keep getting the same error.

The function I'm using to calculate the grains: `grains = calcGrains(ebsd,'threshold',[10*degree 1*degree])`

...and the error message:
**The constructor for class 'grain2d' must return only one output value.
Error in EBSD/calcGrains (line 41)
[grains, newBd] = grain2d(ebsd,V,F,I_DG,I_FD,A_Db,varargin{:});**

Would you know why I keep getting this error? The script runs smoothly in Mtex 5.1, but I wanted to upgrade for a newer version to use the new features :) 

Thanks a lot.
Rhander.

**Comment by kilir** [comment]
Hi Rhander,
I can't reproduce this problem on the recent version, and line 41 in your error message in calcGrains isn't the same line as in the most recent version, so I guess for some reason you are calling the wrong calcGrains. What file opens when you type `edit calcGrains` ?
Cheers,
Rüdiger

**Reply by rtaufner** [reply]
Hi Rüdiger, 
See below the file that opened when I typed `edit calcGrains`.

[calcGrains_5.5_beta6.txt](https://github.com/mtex-toolbox/mtex/files/5612294/calcGrains_5.5_beta6.txt)

I'm using a relatively old version of Matlab (2015b). I don't know if this could be causing the problem...

Thanks for your help.
Cheers, 
Rhander.

**Reply by kilir** [reply]
Within Matlab, the file opens in the editor. Usually at the top bar, it tells you where this file is located. Is this the `calcGrains` you have on disk?
Cheers,
Rüdiger

**Reply by rtaufner** [reply]
Yes, it is. I just copy-pasted in the .txt file the information that popped up in the editor top par when I typed `edit calcGrains` in the command window. See below:
<img width="1280" alt="Screen Shot 2020-11-29 at 11 52 06" src="https://user-images.githubusercontent.com/75056142/100545282-d0a60d00-3239-11eb-82c6-aa4095aafa94.png">

Cheers, 
Rhander.

**Reply by kilir** [reply]
Yes, you are right, there's just changes in the documentation between the beta6 and the 5.5 release version and as far as I can see, I don't see any other problem in relation to this.
Any chance to try a more recent version of Matlab?

Cheers,
Rüdiger

**Reply by rtaufner** [reply]
Hi Rüdiger,
I'll check with my University if they can provide me a newer version of Matlab and try to run the script again.

Thank you very much for your help.
Cheers, 
Rhander.

**Comment by bjoerneske** [comment]
Well as suggested by rudiger there is something left from an old version of mtex on your computer. Make sure to de install old versions and remove the folders with the old versions from matlab path. Then install mtex only having the current mtex version on the matlab path.

**Reply by rtaufner** [reply]
I really don't think this is the case here. When I installed the new Mtex version, I made sure that all previous versions were deleted. I even cleaned my trash folder to guarantee that nothing Mtex-related files were left behind.
